### ISSUE TITLE
About gan code

### ISSUE THREAD

**Issue by UtopiaHu** [cszn/KAIR#56]
Hi, there:
     thanks for your great work. I noticed in your code about GAN training in model_gan.py, the parameters in discriminator D are fixed (requires_grad=false ) before back propagating the gradients of Generator G. Is this 'fix operation' necessary? Thank you.

**Comment by cszn** [comment]
See also https://github.com/xinntao/BasicSR/blob/28883e15eedc3381d23235ff3cf7c454c4be87e6/basicsr/models/srgan_model.py#L84
